### PR TITLE
fix for bug 507

### PR DESCRIPTION
**Pull request by turk-jk**
When video is created from camera the code creates `YPMediaVideo` without asset. please check `YYPPickerVC.swift` line 78-80`. Then we YPMediaVideo after trimming and change cover in `VideoFiltersVC` line 138-139`  and it will fail if there is no asset as we are check of asset is not nil. `let asset = self?.inputVideo.asset`.

### CODE REVIEW

**Comment by turk-jk** [comment]
fix for bug #507

File: Source/Filters/Video/YPVideoFiltersVC.swift [hunk 1]
```diff
@@ -134,13 +134,12 @@ public class YPVideoFiltersVC: UIViewController, IsMediaFilterVC {
                 switch session.status {
                 case .completed:
                     DispatchQueue.main.async {
-                        if let coverImage = self?.coverImageView.image,
-                            let asset = self?.inputVideo.asset {
-                            let resultVideo = YPMediaVideo(thumbnail: coverImage, videoURL: destinationURL, asset: asset)
+                        if let coverImage = self?.coverImageView.image{
```
Comment (NikKovIos) [inline Source/Filters/Video/YPVideoFiltersVC.swift]:
```suggestion
                        if let coverImage = self?.coverImageView.image {
```

**Comment by latheesh9796** [comment]
I am facing this issue and I tried updating and reinstalling the pod today and I still find that this change hasn't been updated in the cocoa-pod library. Can you look into this @NikKovIos :) Thanks

**Comment by NikKovIos** [comment]
@latheesh9796 yes, it's not released at cocoapods. I don't have access rights to trunk it to the main cocoapods repo, only @s4cha has it 🤷🏻‍♂️

**Comment by dathu7728** [comment]
> When video is created from camera the code creates `YPMediaVideo` without asset. please check `YYPPickerVC.swift` line 78-80`. Then we YPMediaVideo after trimming and change cover in `VideoFiltersVC` line 138-139` and it will fail if there is no asset as we are check of asset is not nil. `let asset = self?.inputVideo.asset`.


asset getting nil @turk-jk @NikKovIos 

Please check this issue Mr. @NikKovIos , @turk-jk [](url)

**Pull request merged**